Sources/Rendering/OpenGL/ImageResliceMapper/index.js

```diff
@@ -0,0 +1,1288 @@
+        'normalWCVSOutput = normalWC;',
+        'vertexWCVSOutput = vertexWC;',
+      ]);
+    }
+    VSSource = vtkShaderProgram.substitute(
+      VSSource,
+      '//VTK::PositionVC::Impl',
+      posVCVSImpl
+    ).result;
+    VSSource = vtkShaderProgram.substitute(VSSource, '//VTK::Camera::Dec', [
+      'uniform mat4 MCPCMatrix;',
+      'uniform mat4 MCVCMatrix;',
+    ]).result;
+    let posVCFSDec = [];
+    if (slabThickness > 0.0) {
+      posVCFSDec = posVCFSDec.concat([
+        'varying vec3 normalWCVSOutput;',
+        'varying vec4 vertexWCVSOutput;',
+      ]);
+    }
+    FSSource = vtkShaderProgram.substitute(
+      FSSource,
+      '//VTK::PositionVC::Dec',
+      posVCFSDec
+    ).result;
+    shaders.Vertex = VSSource;
+    shaders.Geometry = GSSource;
+    shaders.Fragment = FSSource;
+  };
+
+  function isVectorAxisAligned(n) {
+    vtkMath.normalize(n);
+    const tmpN = [0, 0, 0];
+    for (let i = 0; i < 3; ++i) {
+      vec3.zero(tmpN);
+      tmpN[i] = 1.0;
+      const dotP = vtkMath.dot(n, tmpN);
+      if (dotP < -0.999 || dotP > 0.999) {
+        return [true, i];
+      }
+    }
+    return [false, 2];
+  }
+
+  publicAPI.updateResliceGeometry = () => {
+    let resGeomString = '';
+    const image = model.currentInput;
+    const imageBounds = image?.getBounds();
+    // Orthogonal slicing by default
+    let orthoSlicing = true;
+    let orthoAxis = 2;
+    if (model.renderable.getSlicePolyData()) {
+      resGeomString = resGeomString.concat(
+        `PolyData${model.renderable.getSlicePolyData().getMTime()}`
+      );
+    } else if (model.renderable.getSlicePlane()) {
+      resGeomString = resGeomString.concat(
+        `Plane${model.renderable.getSlicePlane().getMTime()}`
+      );
+      if (image) {
+        resGeomString = resGeomString.concat(`Image${image.getMTime()}`);
+      }
+      // Check to see if we can bypass oblique slicing related bounds computation
+      [orthoSlicing, orthoAxis] = isVectorAxisAligned(
+        model.renderable.getSlicePlane().getNormal()
+      );
+    } else {
+      // Create a default slice plane here
+      const plane = vtkPlane.newInstance();
+      plane.setNormal(0, 0, 1);
+      let bds = [0, 1, 0, 1, 0, 1];
+      if (image) {
+        bds = imageBounds;
+      }
+      plane.setOrigin(bds[0], bds[2], 0.5 * (bds[5] + bds[4]));
+      model.renderable.setSlicePlane(plane);
+      resGeomString = resGeomString.concat(
+        `Plane${model.renderable.getSlicePlane()?.getMTime()}`
+      );
+      if (image) {
+        resGeomString = resGeomString.concat(`Image${image.getMTime()}`);
+      }
+    }
+
+    if (!model.resliceGeom || model.resliceGeomUpdateString !== resGeomString) {
+      if (model.renderable.getSlicePolyData()) {
+        model.resliceGeom = model.renderable.getSlicePolyData();
+      } else if (model.renderable.getSlicePlane()) {
+        const bounds = image ? imageBounds : [0, 1, 0, 1, 0, 1];
+        if (!orthoSlicing) {
```

Comment:
If there isn't much performance penalty for the general case, it doesn't make sense to add a special case for `orthoSlicing`. It just adds to the complexity of the code and more ways to introduce bugs in the future.
For implicit plane, it would also be easier to just use a larger plane and discard the fragments in the shader that are out-of-bounds texture coordinates.

Reply:
There is a small performance overhead with oblique slicing with an implicit plane - primarily in creating a AABB cropped plane. That's why this change helps. For out-of-bounds tcoords, fragments get a special background color